### PR TITLE
fs: unify Write and Replace(id)

### PR DESCRIPTION
**Pull request by barraguda**
leads to better cleanup of old files written/overwritten by vfs, also better interface.

### CODE REVIEW

File: src/vfs.rs [hunk 1]
```diff
@@ -806,7 +806,7 @@ async fn match_request(
                             message: Message::Request(Request {
                                 inherit: true,
                                 expects_response: Some(5), // TODO evaluate
-                                ipc: Some(serde_json::to_string(&FsAction::Write).unwrap()),
+                                ipc: Some(serde_json::to_string(&FsAction::Write(None)).unwrap()),
```
Comment (barraguda) [inline src/vfs.rs]: changed these to write(none) for now to compile, but we should probably maybe check for existing here? @hosted-fornet

Comment (nick1udwig) [reply src/vfs.rs]: Yeah, I think you are right. Do you understand this code well enough to make this change or should I push a commit here/elsewhere?

Comment (barraguda) [reply src/vfs.rs]: if you have time to push a commit here, would be great! @hosted-fornet

Comment (nick1udwig) [reply src/vfs.rs]: Shoot, forgot about this @bitful-pannul my bad

Comment (nick1udwig) [reply src/vfs.rs]: Added 9014ff3 @bitful-pannul

File: src/vfs.rs [hunk 2]
```diff
@@ -782,22 +782,30 @@ async fn match_request(
...
+                            match vfs.path_to_key.remove(&full_path) {
+                                None => None,
+                                Some(key) => {
+                                    let Key::File { id: hash } = key else {
+                                        panic!("");
+                                    };
+                                    Some(hash)
+                                }
+                            }
+                            // vfs.key_to_entry.remove(&old_key);
```
Comment (barraguda) [inline src/vfs.rs:806]: @hosted-fornet added the same thing for files within a zip directory too, but what's this key_to_entry removal? in the case of an old file lingering at path?

Comment (nick1udwig) [reply src/vfs.rs:806]: Take a look at the data structures and think about it: https://github.com/uqbar-dao/uqbar/blob/82b6baa573067127685c4181683d8fa33bd0f864/src/vfs.rs#L15-L21

Ask me again if you still can't figure it out. This is a pretty important architectural feature of the vfs (hint as to why: https://github.com/uqbar-dao/uqbar/blob/82b6baa573067127685c4181683d8fa33bd0f864/src/vfs.rs#L43-L44)

Comment (barraguda) [reply src/vfs.rs:806]: oh yes I see, but it's not needed here as it's the same key to entry as before right, so unnecessary print?

**Pull request merged**